### PR TITLE
Run representer automatically

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/commands/iteration/representation/process.rb [hunk 1]
```diff
@@ -46,6 +46,10 @@ def call
         rescue StandardError
           iteration.analysis_exceptioned!
         end
+
+        # TODO: Mark iteration as represented and broadcast
```
Comment (iHiD) [inline app/commands/iteration/representation/process.rb]: Can you remove the TODO's and add a test for this pls?

File: app/commands/iteration/analysis/process.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ def call
 
         # TODO: Mark iteration as analyzed and broadcast
         # it here, when we've decided how that works
-        # iteration.broadcast!
+        iteration.broadcast!
```
Comment (iHiD) [inline app/commands/iteration/analysis/process.rb:41]: Can you remove the TODO's and add a test for this pls?

File: app/commands/iteration/create.rb [hunk 3]
```diff
@@ -67,13 +67,13 @@ def init_services
       [
         Thread.new do
           Iteration::TestRun::Init.(iteration_uuid, solution.track.slug, solution.exercise.slug, s3_uri)
+        end,
+        Thread.new do
+          Iteration::Analysis::Init.(iteration_uuid, solution.track.slug, solution.exercise.slug, s3_uri)
```
Comment (iHiD) [inline app/commands/iteration/create.rb:72]: Test pls.

File: app/commands/iteration/create.rb [hunk 4]
```diff
@@ -67,13 +67,13 @@ def init_services
...
+        end,
+        Thread.new do
+          Iteration::Representation::Init.(iteration_uuid, solution.track.slug, solution.exercise.slug, s3_uri)
```
Comment (iHiD) [inline app/commands/iteration/create.rb:75]: Test pls.

File: app/commands/tooling_job/process.rb [hunk 5]
```diff
@@ -24,6 +24,20 @@ def call
           "Nothing to report", # TOOD
           attrs["result"]
         )
+      when "representer"
```
Comment (iHiD) [inline app/commands/tooling_job/process.rb:27]: Test pls.

File: app/commands/iteration/create.rb (hunk 3 above)
Comment (ErikSchierboom) [reply app/commands/iteration/create.rb:72]: It looks like there already is a test for this: https://github.com/exercism/v3-website/blob/master/test/commands/iteration/create_test.rb#L22

File: app/models/iteration/analysis.rb [hunk 6]
```diff
@@ -20,15 +20,15 @@ def ops_errored?
   end
 
   def approved?
-    status == :approved
+    status == :approve
```
Comment (ErikSchierboom) [inline app/models/iteration/analysis.rb:23]: The statuses used did not match [the analyzer interface](https://github.com/exercism/automated-analysis/blob/master/docs/analyzers/interface.md#output-format).

File: test/commands/iteration/analysis/process_test.rb [hunk 7]
```diff
@@ -60,4 +60,14 @@ class Iteration::Analysis::ProcessTest < ActiveSupport::TestCase
 
     assert iteration.reload.analysis_exceptioned?
   end
+
+  test "broadcast" do
+    iteration = create :iteration
+    data = { 'status' => :approve, 'comments' => [] }
+
+    IterationChannel.expects(:broadcast!).with(iteration)
+    IterationsChannel.expects(:broadcast!).with(iteration.solution)
```
Comment (ErikSchierboom) [inline test/commands/iteration/analysis/process_test.rb:69]: Normally I would have separate tests for the two broadcasts, but I've found that the Ruby error message is so clear (exact expectation that failed) that I think combining them is okay.

File: test/commands/iteration/create_test.rb [hunk 8]
```diff
@@ -21,7 +20,6 @@ class Iteration::CreateTest < ActiveSupport::TestCase
     Iteration::TestRun::Init.expects(:call)
     Iteration::Analysis::Init.expects(:call)
     Iteration::Representation::Init.expects(:call)
-    ToolingJob::Create.expects(:call)
```
Comment (ErikSchierboom) [inline test/commands/iteration/create_test.rb:24]: I had to disable this to get the test passing, but I didn't understand at first why I had to do that. The `Iteration::TestRun::Init` call (and the other two) all call `ToolingJob::Create.` in their respective `.call` methods, but it looks like they were not registering. Turns out that `Iteration::TestRun::Init.expects(:call)` causes this. Is it okay to just remove this expectation?

File: test/commands/tooling_job/process_test.rb [hunk 9]
```diff
@@ -39,4 +39,82 @@ class ToolingJob::ProcessTest < ActiveSupport::TestCase
 
     ToolingJob::Process.(id)
   end
+
+  test "proxies to analyzer" do
+    id = SecureRandom.uuid
```
Comment (ErikSchierboom) [inline test/commands/tooling_job/process_test.rb:44]: For now, I've just copy/pasted/modified the existing test runner test. Let me know if you'd like me to use a `setup` method.

File: app/models/iteration/analysis.rb (hunk 6 above)
Comment (iHiD) [reply app/models/iteration/analysis.rb:23]: I think my plan was to change the interface 🙂 

Let's chat about this at some point soon. Certainly `refer_to_mentor` doesn't make sense in v3, but I also wonder about how statuses and comments should work generally.

File: test/commands/iteration/analysis/process_test.rb (hunk 7 above)
Comment (iHiD) [reply test/commands/iteration/analysis/process_test.rb:69]: Yeah - I agree entirely.

File: test/commands/iteration/create_test.rb (hunk 8 above)
Comment (iHiD) [reply test/commands/iteration/create_test.rb:24]: Yeah, I think that's fine now :)

File: test/commands/tooling_job/process_test.rb (hunk 9 above)
Comment (iHiD) [reply test/commands/tooling_job/process_test.rb:44]: No, this is fine for now.

**Comment by iHiD** [comment]
Sweet. Nice work. This is great. I'll merge it now. I think we'll want to revert the analyzer statuses bit further down the line, but until that spec formally change, let's keep it all unified :)

**Pull request merged**